spec/src/main/asciidoc/app-programming-model.adoc

```diff
@@ -504,3 +504,24 @@ The implementation must produce the _vendor_ `MetricRegistry` when the `Registry
 @RegistryType(type=MetricRegistry.Type.VENDOR)
 MetricRegistry vendorRegistry;
 ----
+
+[[pgm-metadata]]
+==== Metadata
+
+Metadata is an interface to construct an immutable object. The object can be built via a `MetadataBuilder` with a fluent api.
```

Comment:
Would be good to say something in this introductory sentence to explain that Metadata is used to provide immutable information about a metric at registration time.